### ISSUE TITLE
ENH: Rework OverviewBar

### ISSUE THREAD

**Issue by marsipu** [mne-tools/mne-qt-browser#27]
> The z-score mode more or less is a minimap I think.

agreed, it has the function of "see at a glance where you are and where interesting features are, and be able to click-drag the viewbox to move around".  But I think it's a bit less intuitive to interpret than a non-transformed minimap would be; and both are far superior to the current non-zscore option.

> So maybe only showing the bottom thing when in z-score mode

agree, unless a more informative non-zscore minimap can be devised.

> and using a vertical one to the just inside or outside of the right scrollbar when in non-zscore mode

If we're going to add something like this, I don't see why we wouldn't have it visible all the time?  Since it won't need to be clickable (there's a separate scrollbar) it could really be just a few pixels thick.

_Originally posted by @drammock in https://github.com/mne-tools/mne-qt-browser/issues/20#issuecomment-969114880_

**Comment by marsipu** [comment]
Regarding the Minimap (which would just zscore now if we moved channels-mode away) it might be hard to identify the **markers for events, annotations and bad channels** with the fidgety background.
#### "channels"-mode
![grafik](https://user-images.githubusercontent.com/46245704/141826712-ef395c51-bf8b-478f-8caa-0eb143e9fb52.png)
#### "zscore"-mode
![grafik](https://user-images.githubusercontent.com/46245704/141826755-d948e821-a7c7-47f9-a13b-bebcad8b20d3.png)

To see those markers, channels-mode or no background at all are probably better. Maybe instead of removing "channels"-mode completely, we add a third option for no background. And the **thin line next to the vertical scrollbar** will be implemented additionally.

**Comment by marsipu** [comment]
### Further ideas?
Do you have further ideas on how to use the background of the OverviewBar/Minimap? (e.g. additional statistical metrics?)

**Comment by larsoner** [comment]
Ohhhh so there *is* information in the "channels" mode, namely the event markers (and maybe annotations?). In that case maybe we should hide it when there are no annotations or event markers only... not sure.

**Comment by marsipu** [comment]
> Ohhhh so there _is_ information in the "channels" mode, namely the event markers (and maybe annotations?). In that case maybe we should hide it when there are no annotations or event markers only... not sure.

No, the markers are independently drawn from the "channels" or "zscore" background. 
I was just thinking when "zscore" will be the only background available without "channels", that would make the markers hard to read.

**Comment by larsoner** [comment]
> No, the markers are independently drawn from the "channels" or "zscore" background.

My point is more that I was wrong above when I said that there wouldn't be any horizontal information in channels mode. There is -- namely the markers (if present). So maybe instead of deciding if we should show the bottom bar based on whether in channels or zscore mode (as I suggested before), we should only hide the bottom bar if in channels mode *and* there are no markers to show

**Comment by marsipu** [comment]
I understand. The users can toggle the OverviewBar with pressing "o" if it takes too much space for them.

**Comment by marsipu** [comment]
> I think it's worth saying that for all-EEG data it's a bit weird to have a big black box down there. Even if there are some annotations showing, it seems a bit odd to put them on a black background.

So in this case you could just as well have no background, right?

> Which I think is another vote for having color-that-indicates-channel-type being a thin rectangle along the vertical scrollbar, and not having it in that overview box.

I am also +1 on that, this thin vertical indicator would also still enable a bit of overview when the OverviewBar is toggled off.

**Comment by drammock** [comment]
> So in this case you could just as well have no background, right?

yes, and further I think that even in cases with multiple channel types, there should be no background (because that info will be available in the new thin rectangle)

**Comment by sappelhoff** [comment]
- I like the idea of displaying channel type information in the vertical scrollbar
- I would like having a "no background" (white/transparent) overview-mode which just shows the annotations ... so pretty much like the "channels" overview mode for all-EEG channels now, but only that it's not black but white for better visibility
- I remember using the FieldTrip `ft_rejectvisual` function some years ago and liked it ... perhaps studying that will also show a few nice options for the overview mode statistics: https://www.fieldtriptoolbox.org/tutorial/visual_artifact_rejection/

**Comment by marsipu** [comment]
I try to implement some of this before 0.2-release, probably won't manage everything

**Comment by aadityasinha-dotcom** [comment]
Hey, I want to work on this issue for GSOC'22.

**Comment by Tasfia-Ara** [comment]
Is this issue still open for GSoC 2025? If so I want to contribute
